### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by looh16**
**Add a method called current_user in  ApplicationController** 
**Add forms to perform the following functions:**

1. Add a Post on behalf of the current_using the current_user
2. Add a comment on behalf of the current_user
3. Allow Users to add likes to Posts.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -2,8 +2,15 @@
...
-  get "users/:id/posts" => "posts#index", as: "user_post"
-  get "users/:id/posts/:id" => "posts#show", as: "post_detail"
+  resources :users do 
+    resources :posts do
+    end
+  end 
+
+  resources :posts do
+    resources :comments 
+    resources :likes
```
Comment (codepantha) [inline config/routes.rb]: - [ ] Please let's endeavour to only set up routes which we will actually be utilizing. As you can see in the attached image below, doing `resources: likes` sets up a bunch of likes related routes which we have no use of. For the purpose of this milestone, we only need to be able to POST a like. Please only set up a `POST /posts/:post_id/likes` so users can like a post 👍 
image

File: app/views/likes/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<div class= "container d-flex flex-column mt-4">
+
+<%= form_with(model: [@post, @like], local: true) do |form| %>
+  <div class="mb-3">
+    <%= form.submit "Save", class: "btn btn-success btn-block" %>
+  </div>
+<% end %>
+
+</div>
```
Comment (codepantha) [inline app/views/likes/new.html.erb]: - [ ] A new page that contains a button simply to like a post is not necessary. Kindly implement the like functionality on the Post `show` page without taking the user to this page 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @looh16 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
Hi @looh16, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights :green_circle: 

- Correct Gitflow :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 
- No Linter error :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by looh16** [comment]
Thanks!

**Pull request merged**